libarchive/archive_write_add_filter_zstd.c

```diff
@@ -315,7 +314,10 @@ static int
 drive_compressor(struct archive_write_filter *f,
     struct private_data *data, int finishing, const void *src, size_t length)
 {
-	ZSTD_inBuffer in = (ZSTD_inBuffer) { src, length, 0 };
+	ZSTD_inBuffer in;
+	in.src = src;
+	in.size = length;
+	in.pos = 0;
 
```

Comment:
Like the `out` instance above - please use the more common one of `{}` `{0}` or `memset`.